inst/include/dust/gpu/rng.hpp

```diff
@@ -0,0 +1,144 @@
+#ifndef DUST_GPU_RNG_HPP
+#define DUST_GPU_RNG_HPP
+
+#include <dust/gpu/xoshiro.hpp>
+#include <dust/gpu/distr/binomial.hpp>
+#include <dust/gpu/distr/normal.hpp>
+#include <dust/gpu/distr/poisson.hpp>
+#include <dust/gpu/distr/uniform.hpp>
+
+namespace dust {
+
+struct RNGptr {
+  uint64_t* state_ptr;
+  unsigned int state_stride;
+  unsigned int particle_stride;
+};
+
+// Read state from global memory
+template <typename T>
+__device__
+rng_state_t<T> loadRNG(RNGptr& rng_state, int p_idx) {
+  rng_state_t<T> state;
+  for (int i = 0; i < XOSHIRO_WIDTH; i++) {
+    state.s[i] = rng_state.state_ptr[p_idx * rng_state.particle_stride +
+                                      i * rng_state.state_stride];
+  }
+  return state;
+}
+
+// Write state into global memory
+template <typename T>
+__device__
+void putRNG(rng_state_t<T>& rng, RNGptr& rng_state, int p_idx) {
+  for (int i = 0; i < XOSHIRO_WIDTH; i++) {
+    rng_state.state_ptr[p_idx * rng_state.particle_stride +
+                        i * rng_state.state_stride] = rng.s[i];
+  }
+}
+
+template <typename real_t>
+class pRNG { // # nocov
```

Comment:
Still want `nocov`?

Reply:
This was needed to avoid a false negative in the covr output (in the CPU version - this file is invisible to coverage checking as it's not compiled into the package binary). I've no idea why it was needed though